File: MS.NET/Sarvam Backend/Models/UserCredentials.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarvam.Models
{
    [Table("user_credentials")]
    public class UserCredentials
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        [Column("PRN")] // Map to the 'PRN' column in the database
        public long PRN { get; set; }

        [Required]
        [Column("Password")] // Map to the 'Password' column
        [StringLength(20)]
        public string Password { get; set; }

        [Required]
        [Column("role")] // Map to the 'role' column
        public string Role { get; set; } = "Student"; // Default role is Student

        [Required]
        [Column("status")] // Map to the 'status' column
        public string Status { get; set; } = "Active"; // Default status is Active

        [Required]
        [Column("created_at")] // Map to the 'created_at' column
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        [Required]
        [Column("updated_at")] // Map to the 'updated_at' column
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }
}

File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/UserCredentials.java
package com.cdac.sarvam.model;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "user_credentials")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserCredentials {

    @Id
    private Long prn;

    @Column(name = "Password", nullable = false)
    private String password;
    
    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;
    
    @Column(name = "email", nullable = false)
    private String email;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private java.sql.Timestamp createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private java.sql.Timestamp updatedAt;

    public enum Role {
        Admin, Student, Instructor, TPO, MoCo, CoCo
    }

    public enum Status {
        Active, Inactive, Suspended, Deleted
    }
}